app/controllers/application_controller.rb

```diff
@@ -5,6 +5,103 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+
+  get '/' do
+    erb :'index'
+  end
+
+
+  get '/signup' do
+    erb :'users/create_user'
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:password] == "" || params[:email] == ""
+      redirect '/failure'
+    else
+      @user =User.create(username: params[:username], email: params[:email], password: params[:password])
+      session[:user_id] = @user.id
+      redirect '/tweets'
+    end
+  end
+
+  get '/login' do
+    erb :'users/login'
+  end
+
+  post "/login" do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_id] = @user.id
+      redirect to "/tweets"
+    else
+      redirect to "/failure"
+    end
+  end
+
+    get '/tweets' do
+      erb :'tweets/tweets'
+    end
+
+    get "/logout" do
+    session.clear
+    redirect "/"
+  end
+
+  get '/tweets/new' do
+    erb :'tweets/create_tweet'
+  end
+
+  post '/tweets' do
+    if !logged_in?
+      redirect to "/failure"
+    else
+      tweet = current_user.tweets.build(params)
+      tweet.save
+    end
+    redirect to '/tweets'
+  end
+
+  get '/tweets/:id' do
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/show'
+    end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find(params[:id])
+    erb :'tweets/edit'
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (not just that any user is logged in) before allowing them to edit?